### PR TITLE
Add support for VERSION field type in SQL and EQL

### LINKED ISSUE #83375: EQL support for fields of type version
EQL doesn't support `version` type fields e.g. `sequence by agent.version` where `agent.version` is of type `version` fails with 

```
verification_exception: Found 1 problem\nline 2:38: Cannot use field [agent.version] with unsupported type [version]
```

The `version` field type was introduced in 7.10 https://github.com/elastic/elasticsearch/pull/59773

### PR DESCRIPTION
**Pull request by luigidellaquila**
Re-submitting #85502, reverted after a CI failure (error on serialization of ConstantProcessor containing a Version object).

The problem was reported by a randomized test, probably it was not identified during the review phase because it's unlikely to happen on CSV spec test suite, due to relatively small result sets.

The applied fix consists in letting ConstantProcessor manage Version objects as a special case and serialize them directly, without relying on StreamInput/Output generic value serialization.
A more natural solution would have been to let StreamInput/Output manage Version objects natively, but it was discarded because Version is in x-pack and cannot be directly referenced/instantiated in Server module (unless we move it and change its license).

Fixes https://github.com/elastic/elasticsearch/issues/83375

### CODE REVIEW

File: docs/changelog/85502.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+pr: 85502
```
Comment (Luegg) [inline docs/changelog/85502.yaml:1]: this file is no longer needed

**Review by Luegg — APPROVED** [review]
Looks generally good to me. But can you give an example of the tests that failed?

**Comment by luigidellaquila** [comment]
@Luegg the failing stack trace was the following:

```
org.elasticsearch.xpack.sql.qa.single_node.JdbcCsvSpecIT > test {version.testCastConstantToVersion2} FAILED
    java.sql.SQLException: Server sent bad type [illegal_argument_exception]. Original type was [can not write type [class org.elasticsearch.xpack.versionfield.Version]]. [java.lang.IllegalArgumentException: can not write type [class org.elasticsearch.xpack.versionfield.Version]
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.io.stream.StreamOutput.writeGenericValue(StreamOutput.java:828)
        at org.elasticsearch.xpack.ql.expression.gen.processor.ConstantProcessor.writeTo(ConstantProcessor.java:68)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.io.stream.StreamOutput.writeNamedWriteable(StreamOutput.java:1097)
        at org.elasticsearch.xpack.ql.execution.search.extractor.ComputingExtractor.writeTo(ComputingExtractor.java:56)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.io.stream.StreamOutput.writeNamedWriteable(StreamOutput.java:1097)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.io.stream.StreamOutput.writeNamedWriteableList(StreamOutput.java:1203)
        at org.elasticsearch.xpack.sql.execution.search.SearchHitCursor.writeTo(SearchHitCursor.java:80)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.io.stream.StreamOutput.writeNamedWriteable(StreamOutput.java:1097)
        at org.elasticsearch.xpack.sql.session.Cursors.encodeToString(Cursors.java:82)
        at org.elasticsearch.xpack.sql.session.Cursors.encodeToString(Cursors.java:73)
        at org.elasticsearch.xpack.sql.plugin.TransportSqlQueryAction.createResponse(TransportSqlQueryAction.java:225)
        at org.elasticsearch.xpack.sql.plugin.TransportSqlQueryAction.createResponseWithSchema(TransportSqlQueryAction.java:206)
        at org.elasticsearch.xpack.sql.plugin.TransportSqlQueryAction.lambda$operation$0(TransportSqlQueryAction.java:170)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)
        at org.elasticsearch.xpack.sql.execution.search.SearchHitCursor.handle(SearchHitCursor.java:181)
        at org.elasticsearch.xpack.sql.execution.search.Querier$SearchHitActionListener.handleResponse(Querier.java:616)
        at org.elasticsearch.xpack.sql.execution.search.Querier$BaseActionListener.onResponse(Querier.java:686)
        at org.elasticsearch.xpack.sql.execution.search.Querier$BaseActionListener.onResponse(Querier.java:665)
        at org.elasticsearch.xpack.sql.execution.search.Querier.lambda$searchWithPointInTime$0(Querier.java:159)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$2.onResponse(ActionListener.java:162)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.client.internal.node.NodeClient$ActionResponseTaskListener.onResponse(NodeClient.java:175)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:171)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.tasks.TaskManager$1.onResponse(TaskManager.java:165)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.ActionListener$RunAfterActionListener.onResponse(ActionListener.java:367)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.AbstractSearchAsyncAction.sendSearchResponse(AbstractSearchAsyncAction.java:722)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.FetchLookupFieldsPhase.run(FetchLookupFieldsPhase.java:75)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:469)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:463)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.ExpandSearchPhase.onPhaseDone(ExpandSearchPhase.java:151)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.ExpandSearchPhase.run(ExpandSearchPhase.java:105)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:469)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:463)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:269)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$2(FetchSearchPhase.java:106)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:115)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:90)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:768)
        at org.elasticsearch.server@8.4.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

It reproduced deterministically with a specific seed:

```
./gradlew ':x-pack:plugin:sql:qa:server:single-node:javaRestTest' --tests "org.elasticsearch.xpack.sql.qa.single_node.JdbcCsvSpecIT" -Dtests.method="test {version.testScriptsOperatorsOrderBy}" -Dtests.seed=8B1768D7955260FA -Dtests.locale=hi -Dtests.timezone=Etc/UCT -Druntime.java=17
```

**Review by astefan — APPROVED** [review]
LGTM

**Review by bpintea — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM

**Comment by luigidellaquila** [comment]
@elasticmachine update branch

**Pull request merged**